File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if (n<=1) return 1;

        int prev1 = 1;
        int prev2 = 2;
        
        for (int i = 3; i<=n; i++){
            int val = prev1 + prev2; //Similar to Fibbanocci.
            prev1= prev2;
            prev2=val; 
        }
        return prev2;
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n){        //SC     
        if(n==1 || n==2){
            return n;
        }
        int num1 = 1;
        int num2 = 2;
        for(int i=3; i<=n; i++){
            int num3 = num2 + num1;
            num1 = num2;
            num2 = num3;
        }
        return num2;
    }
};